include/core/evaporation_model.h

```diff
@@ -0,0 +1,683 @@
+  {
+    std::fill(jacobian_vector.begin(), jacobian_vector.end(), 0);
+  }
+  /**
+   * @brief momentum_flux Calculates the value of the evaporation recoil pressure.
+   * @param fields_value Value of the various field on which the recoil pressure
+   * may depend.
+   * @return value of the recoil pressure calculated with the fields_value
+   */
+  double
+  momentum_flux(const std::map<field, double> & /*fields_value*/) override
+  {
+    return -n_evaporation * n_evaporation *
+           (liquid_density_inv - ambient_gas_density_inv);
+  }
+
+  /**
+   * @brief vector_momentum_flux Calculates the values of the evaporation recoil
+   * pressure for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param momentum_flux_vector Vectors of the recoil pressure values
+   */
+  void
+  vector_momentum_flux(
+    const std::map<field, std::vector<double>> & /*field_vectors*/,
+    std::vector<double> &momentum_flux_vector) override
+  {
+    const double momentum_flux_value =
+      -n_evaporation * n_evaporation *
+      (liquid_density_inv - ambient_gas_density_inv);
+
+    std::fill(momentum_flux_vector.begin(),
+              momentum_flux_vector.end(),
+              momentum_flux_value);
+  }
+
+  /**
+   * @brief momentum_flux_jacobian Calculates the jacobian (the partial derivative)
+   * of the evaporation recoil pressure with respect to a field
+   * @param field_values Value of the various fields on which the recoil pressure
+   * may depend.
+   * @param id Indicator of the field with respect to which the jacobian
+   * should be calculated
+   * @return value of the partial derivative of the recoil pressure with respect
+   * to the field.
+   */
+  double
+  momentum_flux_jacobian(const std::map<field, double> & /*field_values*/,
+                         const field /*id*/) override
+  {
+    return 0.0;
+  }
+
+  /**
+   * @brief vector_momentum_flux_jacobian Calculate the derivative of the
+   * evaporation recoil pressure with respect to a field
+   * @param field_vectors Vector for the values of the fields on which the recoil
+   * pressure may depend.
+   * @param id Identifier of the field with respect to which a derivative should
+   * be calculated
+   * @param jacobian Vector of the value of the derivative of the recoil pressure
+   * with respect to the field id
+   */
+  void
+  vector_momentum_flux_jacobian(
+    const std::map<field, std::vector<double>> & /*field_vectors*/,
+    const field /*id*/,
+    std::vector<double> &jacobian_vector) override
+  {
+    std::fill(jacobian_vector.begin(), jacobian_vector.end(), 0);
+  }
+
+private:
+  const double n_evaporation;
+  const double latent_heat_evaporation;
+  const double ambient_gas_density_inv;
+  const double liquid_density_inv;
+};
+
+/**
+ * @brief EvaporationModelConstant. Class that allows to calculate the
+ * evaporation heat and momentum fluxes when a constant mass flux is
+ * considered.
```

Comment:
This is not a constant mass flux no, it depends on temperature. I think you need to update the documentation here to reflect the model. Some slight copy paste I believe ;)